### ISSUE TITLE
Bug with Static page URL's

### ISSUE THREAD

**Issue by canthis** [rainlab/translate-plugin#286]
##### Expected behavior
Multilingual Static page URL's should be generated uniquely based on page title locales being entered.

##### Actual behavior
Entering page title for another language makes URL the same for all locales so User is now forced to manually enter/edit page URL's for each locale and this could be a pain if there's more than 5 languages and especially for non dev user.

##### Reproduce steps
[https://youtu.be/jpcnVAu6STM](https://youtu.be/jpcnVAu6STM)

Let's assume we have 2 or more languages: EN(default) and RU
1. Add static page
2. Enter page title for EN, for example - "Contacts"
3. Page URL for all the locales are now - "/contacts"
4. Switch to another language and type in page title, for example - "Контакты" 
5. Page URL's for all the languages are now - "/kontakty"
---
I will assume that this problem occurs because of all the Page URL hidden inputs are pointing to same Input.Preset field - `data-input-preset="[name="viewBag[title]"]"`
```
<input type="text" name="viewBag[url]" id="Form-formPaged41d8cd98f00b204e9800998ecf8427e5984ace7238a3-field-viewBag-url-placeholderField" value="" placeholder="/" class="form-control" autocomplete="off" maxlength="255" data-input-preset="[name="viewBag[title]"]" data-input-preset-type="url" data-input-preset-closest-parent="form" data-input-preset-prefix-input="input[data-parent-url]">

<button class="btn btn-default ml-btn" data-toggle="dropdown" data-active-locale="" type="button">en</button>

<!-- Individual values -->
<input type="hidden" name="RLTranslate[lv][viewBag][url]" value="/aaa" data-locale-value="lv" data-input-preset="[name="viewBag[title]"]" data-input-preset-type="url" data-input-preset-closest-parent="form" data-input-preset-prefix-input="input[data-parent-url]">
<input type="hidden" name="RLTranslate[ru][viewBag][url]" value="/aaa" data-locale-value="ru" data-input-preset="[name="viewBag[title]"]" data-input-preset-type="url" data-input-preset-closest-parent="form" data-input-preset-prefix-input="input[data-parent-url]">
<input type="hidden" name="RLTranslate[en][viewBag][url]" value="/aaa" data-locale-value="en" data-input-preset="[name="viewBag[title]"]" data-input-preset-type="url" data-input-preset-closest-parent="form" data-input-preset-prefix-input="input[data-parent-url]">
```

##### October build
420 (Same behavior is on build 419)

**Comment by amdad** [comment]
I confirm that.

**Comment by vannut** [comment]
Confirmed

**Comment by nicolagaldiolo** [comment]
Confirmed.

**Comment by lucianomelotti** [comment]
I confirm the problem

**Comment by LukeTowers** [comment]
@amdad @vannut @niclasleonbock @lucianomelotti great, thanks for confirming that. It would be helpful if you guys could take some time to dig into what's causing the issue and provide updates on your progress here. That way we can get this fixed sooner rather than later. Thanks!

**Comment by LukeTowers** [comment]
Also, whenever you see the `Type: Unconfirmed Bug` label, it would be significantly more helpful if you could provide your own summary of the bug with as much detail as possible informing us how it affects you specifically rather than just saying "Confirmed.". A one word addition doesn't really give us anything to work off of.

**Comment by LukeTowers** [comment]
@amdad @vannut @niclasleonbock @lucianomelotti @canthis could somebody try debugging this in more depth and provide a fix via Pull Request?

**Comment by vannut** [comment]
I can't reproduce the bug anymore in the latest version (1.3.2) of the plugin.

**Comment by LukeTowers** [comment]
Closing this unless other information reliably reproducing this comes up.

**Comment by tdolph** [comment]
Where can i get the 1.3.2 version? I can only see 1.2.18?

**Comment by LukeTowers** [comment]
@tdolph see previous comment: https://github.com/rainlab/translate-plugin/issues/286#issuecomment-339734186

**Comment by tdolph** [comment]
Im not familiar with this plugins code but it appears to me that the problem is actually octobers own preset functionality. It sets values of both languages when typing inside title input. As far as i see, there is no problem posting the right values, only with setting those input values. I will dig further and try to make it work.

**Comment by tdolph** [comment]
Been at it for almost a day now but i just cant find the js responsible for overwriting it. Im not very good at js so some help would be really appreciated!

**Comment by LukeTowers** [comment]
@vannut could you help @tdolph find the cause for this issue?

**Comment by iboved** [comment]
I have the same problem.
The reason is all the Page URL hidden inputs are pointing to same Input.Preset field:

`data-input-preset="[name="viewBag[title]"]"`

All fields are generated from the "Title" field.

**Comment by LukeTowers** [comment]
@iboved are you able to come up with a fix?

**Comment by rubenvanerk** [comment]
I have the same issue. Tried to fix it but no success yet.
JS code responsible for filling the preset field is in modules/system/assets/ui/js/input.preset.js

@iboved is correct, the slug field is generated from the overarching input field and not from one of underlaying the hidden lang input fields.

I tried to change the data-input-preset to RLTranslate[<lang>][<fieldname>] but that doesn't seem to work.

Any pointers in the right direction would be nice.
